### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by amit8726** [koalaman/shellcheck#1552]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2148, SC2115, SC2046, SC2143, SC2046, SC2046, SC2154, SC2242, SC2046
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
full_backup() {

if [ ! -d "$BACKUP_DIR_ROOT" ]
then
mkdir "$BACKUP_DIR_ROOT"
fi
if [ ! -d "$BACKUP_DIR" ]
then
mkdir "$BACKUP_DIR"
fi

rm -rf "$BACKUP_DIR"/*
echo $(date '+%Y-%m-%d %H:%M:%S:%s')": Cleanup the backup folder is done! Starting backup" >> "$BACKUP_DIR"/xtrabackup.log

#xtrabackup --backup $SECRET -u root -p --history --compress --slave-info --compress-threads=4 --target-dir=$BACKUP_DIR/FULL
xtrabackup --backup "$SECRET" --history --compress --compress-threads=4 --target-dir="$BACKUP_DIR"/FULL > "$TMPFILE" 2>&1
if [ -z "$(tail -1 "$TMPFILE" | grep 'completed OK!')" ]
then

echo $(date '+%Y-%m-%d %H:%M:%S:%s')": Error! Backup Failed.." >> "$BACKUP_DIR"/xtrabackup.log
ERRMSG="DailyFullXtraBackup Job on server $HOSTNAME has been failed "

echo $(date '+%Y-%m-%d %H:%M:%S:%s')": sending alert for failure" >> "$BACKUP_DIR"/xtrabackup.log
echo -e "$ERRMSG" | mail -r "root@abc.com" -s "backup failed" "$emails"
rm "$TMPFILE"
touch test.txt
exit -1
fi
echo $(date '+%Y-%m-%d %H:%M:%S:%s')": Backup Done!" >> "$BACKUP_DIR"/xtrabackup.log
}

```

#### Here's what shellcheck currently says:
```
Line 1	SC2148: Tips depend on target shell and yours is unknown. Add a shebang.
Line 12	SC2115: Use "${var:?}" to ensure this never expands to /* .
Line 13	SC2046: Quote this to prevent word splitting.
Line 17	SC2143: Use ! grep -q instead of comparing output with [ -z .. ].
Line 20	SC2046: Quote this to prevent word splitting.
Line 23	SC2046: Quote this to prevent word splitting.
Line 24	SC2154: emails is referenced but not assigned.
Line 27	SC2242: Can only exit with status 0-255. Other data should be written to stdout/stderr.
Line 29	SC2046: Quote this to prevent word splitting.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

**Comment by koalaman** [comment]
These warnings all seem legitimate, so I'm assuming it was filed by accident. Please reopen if this is not the case.
